File: ArrHash/217/main.go
package main

func containsDuplicate(nums []int) bool {
	//making a map in go where num is the key and bool is the value
	hash := make(map[int]bool)
	for _, num := range nums {
		// if key is present in the map and the value is true then return true
		if _, ok := hash[num]; ok {
			return true
		}
		//otherwise the value for that key is set to true
		hash[num] = true
	}
	return false
}

File: ArrHash/217/main.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
            #set creation for previous numbers
            prev = set()
            for num in nums:
                #if number was already added to the set, return True
                if num in prev:
                    return True
                #otherwise add the number
                prev.add(num)
            return False